hypothesis/client
# Modify `update-changelog.js` to generate the list of changes itself

In preparation for executing the client release process entirely on
Jenkins, make update-changelog.js automatically update CHANGELOG.md with
a list of changes since the most recent tag, instead of requiring a
developer to do this manually.

This will lose the advantages of having a changelog written by humans, but
we primarily use other channels nowadays to notify staff and users about
Hypothesis product changes.

Why continue to have a CHANGELOG.md at all? Because I think it is quite useful to have a nicely formatted list of changes in the GitHub releases page. There are also other tools and services (eg. [Yarn package search](https://yarnpkg.com/en/package/hypothesis)) which make use of it.

----

**Testing**

1. Set your `GITHUB_TOKEN` env var locally to a personal access token for GitHub. This is optional but the next step may hit API rate limits if you don't do this.
2. Run `scripts/update-changelog.js`.

You should find this modifies CHANGELOG.md with a diff like this:

```diff
diff --git a/CHANGELOG.md b/CHANGELOG.md
index bc7d4135..9681e99d 100644
--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -4,6 +4,17 @@ Entries in this change log follow the format suggested at http://keepachangelog.
 
 # Change Log
 
+## [1.93.0] - 2018-10-16
+
+### Changed
+
+- Support serving the client over SSL in development
+  [#785](https://api.github.com/repos/hypothesis/client/pulls/785)
+
+- Update API route definitions used by tests
+  [#786](https://api.github.com/repos/hypothesis/client/pulls/786)
+
+
 ## [1.93.0] - 2018-10-09
 
 ### Changed
```

Note that the version number is taken from the _current_ package.json file, so would be a _new_ version number in a real release.
